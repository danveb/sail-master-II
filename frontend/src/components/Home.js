import React from 'react' 
import { Container } from 'react-bootstrap' 

const Home = ({ currentUser }) => {
    const greetUser = (currentUser) => {
        let user = currentUser.firstName[0].toUpperCase().concat(currentUser.firstName.slice(1).concat('!'))
        return user; 
    }

    return (
        <Container className="d-flex align-items-center justify-content-center mt-4">
            {currentUser && (
                <div className="w-100 text-center">
                    <h1>Welcome, {greetUser(currentUser)} </h1>
                    <p>You can go ahead and create a Voyage</p>
                </div>
            )}

            {!currentUser && (
                <div className="w-100 text-center">
                    <h1>Welcome to Sail Master II</h1>
                    <p>Please register if this is your first time!</p>
                    <p>Happy Sailing!</p>
                </div>
            )}
        </Container>
    )
}

export default Home 