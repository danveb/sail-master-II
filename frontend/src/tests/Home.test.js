import React from 'react'
import { render } from '@testing-library/react'
import Home from '../components/Home' 

// smoke test 
it('renders without crashing', () => {
    render(<Home />) 
})

// snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
})