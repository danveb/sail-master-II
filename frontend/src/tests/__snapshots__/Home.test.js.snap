// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center mt-4 container"
  >
    <div
      class="w-100 text-center"
    >
      <h1>
        Welcome to Sail Master II
      </h1>
      <p>
        Please register if this is your first time!
      </p>
      <p>
        Happy Sailing!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
